functionary/ui/forms/scheduled_task.py

```diff
@@ -36,59 +36,83 @@ class ScheduledTaskForm(ModelForm):
         initial="*",
         validators=[month_of_year_validator],
     )
+    function = ModelChoiceField(queryset=Function.objects.all(), required=True)
 
     class Meta:
         model = ScheduledTask
+
+        # NOTE: The order of the fields matters. The clean_<field> methods run based on
+        # the order they are defined in the 'fields =' attribute
         fields = [
             "name",
+            "environment",
             "description",
             "status",
             "function",
             "parameters",
-            "environment",
         ]
 
     def __init__(self, *args, **kwargs):
-        env: Environment = kwargs.pop("env")
+        environment = kwargs.pop("environment", None)
+        is_create = kwargs.pop("is_create", False)
         super().__init__(*args, **kwargs)
-        self.fields["function"].queryset = Function.objects.filter(
-            package__environment=env
-        )
-        self.fields["status"].choices = self._get_status_choices()
+        self._setup_field_choices(is_create)
+        self._update_function_queryset(environment)
         self._setup_field_classes()
 
-    def clean(self):
-        cleaned_data = super().clean()
-        available_functions = Function.objects.filter(
-            package__environment=cleaned_data["environment"]
-        )
-        if cleaned_data["function"] not in available_functions:
+    def clean_function(self):
+        """The function field needs to be declared AFTER the environment field"""
```

Comment:
I had added the clean method for the function just for the assurance that the user was only scheduling a function that they had access to. I removed it since it is redundant